android/java/org/chromium/chrome/browser/crypto_wallet/util/JavaUtils.java

```diff
@@ -70,4 +72,23 @@ public static boolean anyNull(Object... items) {
         }
         return false;
     }
+
+    /**
+     * Returns a combined string with a separator or an empty string. Empty and null values are
+     * ignored e.g. concatStrings(#,a,b,c,"",null) => a#b#c
+     * @param separator to append after each string.
+     * @param items of string values.
+     * @return a combined or empty string.
+     */
+    public static String concatStrings(char separator, String... items) {
+        if (items == null) return "";
+        StringBuilder builder = new StringBuilder();
+        for (String item : items) {
+            if (TextUtils.isEmpty(item)) continue;
+            builder.append(item).append(separator);
+        }
+        // Delete separator after the last value
+        builder.deleteCharAt(builder.length() - 1);
+        return builder.toString();
+    }
```

Comment:
The logic implemented using for-loop for handling string concatenation is flawless and works smoothly. 
I'm adding an alternative implementation leveraging Java streams that I find slightly more readable. Including it just in the interest of enhancing our knowledge, not asking for any further change here. 

```java
    public static String concatStrings(char separator, String... items) {
        if (items == null) {
            return "";
        }
        return Arrays.stream(items)
                .filter(item -> !TextUtils.isEmpty(item))
                .collect(Collectors.joining(String.valueOf(separator)));
    }
```